pkg/cluster/registerWithHive.go

```diff
@@ -0,0 +1,130 @@
+package cluster
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/openshift/installer/pkg/asset/installconfig/azure"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/hiveregistrator"
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+)
+
+func (m *manager) collectDataForHive(ctx context.Context) (*hiveregistrator.WorkloadCluster, error) {
+	if m.hiveClients == nil {
+		m.log.Print("No hive client, skipping step")
+		return nil, nil
+	}
+	m.log.Print("collecting registration data from new cluster")
+
+	cv, err := m.configcli.ConfigV1().ClusterVersions().Get(ctx, "version", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	clusterID := cv.Spec.ClusterID
+
+	namespacename, err := m.hiveClients.UUIDClusterNamespaceName(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: When hive support first party principles we'll need to send both first party and cluster service principles
+	servicePrincipal := azure.Credentials{
+		TenantID:       m.subscriptionDoc.Subscription.Properties.TenantID,
```

Comment:
Check if we need to do `nil` check here: `Subscription` is a pointer.

Reply:
Technically speaking the subscription could be nil, but in that case we'd fail much earlier in the installation process and we don't check for nil in the cases either.
Suggest to leave as it.